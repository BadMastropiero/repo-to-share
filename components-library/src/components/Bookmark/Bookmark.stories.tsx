import {ComponentMeta, ComponentStory} from '@storybook/react';
import {MailEnvelope, PhoneInCall, Share} from '../Icon';

import Bookmark from './Bookmark';

export default {
  title: 'Components/Bookmark',
  component: Bookmark,
  argTypes: {
    href: {
      control: {
        type: 'text',
      },
    },
  },
} as ComponentMeta<typeof Bookmark>;

export const Base: ComponentStory<typeof Bookmark> = args => {
  const {primary, text, href} = args;

  return (
    <Bookmark
      primary={primary}
      text={text}
      beforeIcon={Share}
      href={href}
      // afterIcon={ArrowLeft}
      // onClick={onClick}
    />
  );
};

Base.args = {
  primary: true,
  text: 'Bookmark text',
};

export const MailBookmark: ComponentStory<typeof Bookmark> = () => (
  <Bookmark
    primary
    text="info@tetraimpacts.com"
    beforeIcon={MailEnvelope}
    href="mailto: info@tetraimpacts.com"
  />
);

MailBookmark.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/epHXucNHjke2rES69iOsaS/TetraImpacts?node-id=2555%3A7532&t=Yx1jzZZsTXIGvj6Z-4',
  },
};

export const PhoneBookmark: ComponentStory<typeof Bookmark> = () => (
  <Bookmark primary text="+ 505-7714-1105" beforeIcon={PhoneInCall} href="tel: + 505-7714-1105" />
);

PhoneBookmark.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/epHXucNHjke2rES69iOsaS/TetraImpacts?node-id=2555%3A7538&t=Yx1jzZZsTXIGvj6Z-4',
  },
};

export const BookmarkWithAfterIcon: ComponentStory<typeof Bookmark> = () => (
  <Bookmark primary text="+ 505-7714-1105" afterIcon={PhoneInCall} />
);

export const BookmarkWithBothIcons: ComponentStory<typeof Bookmark> = () => (
  <Bookmark primary text="+ 505-7714-1105" afterIcon={PhoneInCall} beforeIcon={MailEnvelope} />
);
