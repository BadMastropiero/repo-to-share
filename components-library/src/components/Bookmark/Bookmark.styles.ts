import styled, {css} from 'styled-components';
import {DynamicComponent} from '../DynamicComponent';

const StyledBookmark = styled(DynamicComponent)<{
  primary?: boolean;
}>(
  ({
    theme: {
      colors: {primary600, primaryBg},
      transition,
    },
    primary: isPrimary,
  }) => css`
    color: unset;
    text-decoration: none;

    box-sizing: border-box;
    position: relative;
    left: 0;
    top: 0;
    fill: none;
    display: flex;
    flex-direction: row;
    min-height: 80px;

    background-color: ${(isPrimary && primary600) || primaryBg};
    border-radius: 10px;

    transition: ${transition};

    &:hover {
      /* box-shadow: 0px 10px 15px -5px rgba(0, 101, 166, 0.4); */
    }
  `
);

const BmIconContainer = styled.div<{
  primary?: boolean;
}>(
  ({
    theme: {
      colors: {primary: primaryColor, primary100},
      transition,
    },
    primary: isPrimary,
  }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-grow: 0;
    flex-shrink: 0;

    width: 76px;

    transition: ${transition};
    pointer-events: 'none';

    &.before {
      border-right: 4px dashed ${isPrimary ? primaryColor : primary100};
    }

    &.after {
      border-left: 4px dashed ${isPrimary ? primaryColor : primary100};
    }
  `
);

export {BmIconContainer};
export default StyledBookmark;
