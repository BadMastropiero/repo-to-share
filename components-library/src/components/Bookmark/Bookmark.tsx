import React, {useContext} from 'react';
import {ThemeContext} from 'styled-components';

// Interfaces
import BookmarkProps from './Bookmark.interface';

// Styles
import StyledBookmark, {BmIconContainer} from './Bookmark.styles';

// Components
import {Icon} from '../Icon';
import {Typography} from '../Typography';

const Bookmark: React.FC<BookmarkProps> = ({
  text,
  primary,
  beforeIcon,
  afterIcon,
  className,
  href,
  children,
  alwaysShowRibbon,
  ...rest
}) => {
  const {colors} = useContext(ThemeContext);

  const showRightRibbon = afterIcon || alwaysShowRibbon === 'both' || alwaysShowRibbon === 'right';
  const showLeftRibbon = beforeIcon || alwaysShowRibbon === 'both' || alwaysShowRibbon === 'left';
  return (
    <StyledBookmark
      as={StyledBookmark}
      tag={href && 'link'}
      href={href}
      primary={primary}
      className={`ti-bookmark ${className}`}
      {...rest}
    >
      {showLeftRibbon && (
        <BmIconContainer primary={primary} className="ti-bookmark-icon-container before">
          {beforeIcon && (
            <Icon
              icon={beforeIcon}
              primary={!primary}
              width="24px"
              height="24px"
              className="ti-bookmark-icon before"
            />
          )}
        </BmIconContainer>
      )}
      {text && (
        <Typography
          className="ti-bookmark-text"
          customColor={primary ? colors.primaryBg : colors.primary}
          fontSize="20px"
          fontStyle="normal"
          fontWeight="700"
          lineHeight="200%"
          preset="content"
          variant="p"
          style={{
            padding: '0 20px',
            width: '100%',
            alignSelf: 'center',
          }}
        >
          {text}
        </Typography>
      )}
      {children}
      {showRightRibbon && (
        <BmIconContainer primary={primary} className="ti-bookmark-icon-container after">
          {afterIcon && (
            <Icon
              icon={afterIcon}
              primary={!primary}
              width="24px"
              height="24px"
              className="ti-bookmark-icon after"
            />
          )}
        </BmIconContainer>
      )}
    </StyledBookmark>
  );
};

export default Bookmark;
