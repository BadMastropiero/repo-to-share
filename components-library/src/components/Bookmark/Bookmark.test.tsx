import {render, screen} from '@testing-library/react';
import {composeStories} from '@storybook/testing-react';
import '@testing-library/jest-dom';

import * as stories from './Bookmark.stories';

const {Base, MailBookmark} = composeStories(stories);

describe('Bookmark tests', () => {
  test('should render', () => {
    render(<Base />);
    expect(screen.getByText('Bookmark text')).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const tree = render(<MailBookmark />);
    expect(tree).toMatchSnapshot();
  });
});
