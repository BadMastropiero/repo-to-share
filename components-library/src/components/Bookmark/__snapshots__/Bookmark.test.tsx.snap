// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="sc-gswNZR sc-dkrFOg gwPJTp sc-dkrFOg gwPJTp ti-bookmark undefined"
        href="mailto: info@tetraimpacts.com"
      >
        <div
          class="sc-hLBbgP hmQTFX ti-bookmark-icon-container before"
        >
          <svg
            class="sc-bcXHqe cMqtmU ticons-mail_envelope"
            height="24px"
            role="img"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.84995 20.75C3.11662 20.75 2.49162 20.4917 1.97495 19.975C1.45828 19.4583 1.19995 18.8333 1.19995 18.1V5.9C1.19995 5.16667 1.45828 4.54167 1.97495 4.025C2.49162 3.50833 3.11662 3.25 3.84995 3.25H20.15C20.8833 3.25 21.5083 3.50833 22.025 4.025C22.5416 4.54167 22.7999 5.16667 22.7999 5.9V18.1C22.7999 18.8333 22.5416 19.4583 22.025 19.975C21.5083 20.4917 20.8833 20.75 20.15 20.75H3.84995ZM20.15 8.6L12.675 13.15C12.575 13.2167 12.4666 13.2667 12.35 13.3C12.2333 13.3333 12.1166 13.35 12 13.35C11.8833 13.35 11.7666 13.3333 11.65 13.3C11.5333 13.2667 11.425 13.2167 11.325 13.15L3.84995 8.6V18.1H20.15V8.6ZM12 11L20.35 5.9H3.64995L12 11Z"
            />
          </svg>
        </div>
        <p
          class="sc-gswNZR sc-eDvSVe fhyHkD ti-bookmark-text"
          font-size="20px"
          font-style="normal"
          font-weight="700"
          style="padding: 0px 20px; width: 100%; align-self: center;"
        >
          info@tetraimpacts.com
        </p>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="sc-gswNZR sc-dkrFOg gwPJTp sc-dkrFOg gwPJTp ti-bookmark undefined"
      href="mailto: info@tetraimpacts.com"
    >
      <div
        class="sc-hLBbgP hmQTFX ti-bookmark-icon-container before"
      >
        <svg
          class="sc-bcXHqe cMqtmU ticons-mail_envelope"
          height="24px"
          role="img"
          viewBox="0 0 24 24"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.84995 20.75C3.11662 20.75 2.49162 20.4917 1.97495 19.975C1.45828 19.4583 1.19995 18.8333 1.19995 18.1V5.9C1.19995 5.16667 1.45828 4.54167 1.97495 4.025C2.49162 3.50833 3.11662 3.25 3.84995 3.25H20.15C20.8833 3.25 21.5083 3.50833 22.025 4.025C22.5416 4.54167 22.7999 5.16667 22.7999 5.9V18.1C22.7999 18.8333 22.5416 19.4583 22.025 19.975C21.5083 20.4917 20.8833 20.75 20.15 20.75H3.84995ZM20.15 8.6L12.675 13.15C12.575 13.2167 12.4666 13.2667 12.35 13.3C12.2333 13.3333 12.1166 13.35 12 13.35C11.8833 13.35 11.7666 13.3333 11.65 13.3C11.5333 13.2667 11.425 13.2167 11.325 13.15L3.84995 8.6V18.1H20.15V8.6ZM12 11L20.35 5.9H3.64995L12 11Z"
          />
        </svg>
      </div>
      <p
        class="sc-gswNZR sc-eDvSVe fhyHkD ti-bookmark-text"
        font-size="20px"
        font-style="normal"
        font-weight="700"
        style="padding: 0px 20px; width: 100%; align-self: center;"
      >
        info@tetraimpacts.com
      </p>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
